clients/spark/build.sbt

```diff
@@ -24,47 +22,38 @@ def settingsToCompileIn(dir: String) = {
   )
 }
 
-def generateCoreProject(buildType: BuildType) =
-  Project(s"${baseName}-client-${buildType.name}", file(s"target/core-${buildType.name}"))
+lazy val core =
+  (project in file(s"core"))
     .settings(
+      name := "lakefs-spark-client",
       sharedSettings,
       settingsToCompileIn("core"),
-      scalaVersion := buildType.scalaVersion,
       PB.targets := Seq(
         scalapb.gen() -> (Compile / sourceManaged).value / "scalapb"
       ),
-      libraryDependencies ++= Seq("org.rocksdb" % "rocksdbjni" % "6.6.4",
+      libraryDependencies ++= Seq(
+        "org.rocksdb" % "rocksdbjni" % "6.6.4",
         "commons-codec" % "commons-codec" % "1.15",
-        "org.apache.spark" %% "spark-sql" % buildType.sparkVersion % "provided",
         "com.thesamet.scalapb" %% "scalapb-runtime" % scalapb.compiler.Version.scalapbVersion % "protobuf",
-        "org.apache.hadoop" % "hadoop-aws" % buildType.hadoopVersion,
-        "org.apache.hadoop" % "hadoop-common" % buildType.hadoopVersion,
+        "org.apache.hadoop" % "hadoop-aws" % "2.7.7",
+        "org.apache.hadoop" % "hadoop-common" % "2.7.7",
         "org.scalaj" %% "scalaj-http" % "2.4.2",
         "org.json4s" %% "json4s-native" % "3.7.0-M8",
         "com.google.guava" % "guava" % "16.0.1",
         "com.google.guava" % "failureaccess" % "1.0.1",
       )
     )
 
-def generateExamplesProject(buildType: BuildType) =
-  Project(s"${baseName}-examples-${buildType.name}", file(s"target/examples-${buildType.name}"))
+lazy val examples =
+  (project in file(s"examples"))
     .settings(
+      name := "lakefs-spark-example",
       sharedSettings,
       settingsToCompileIn("examples"),
-      scalaVersion := buildType.scalaVersion,
-      libraryDependencies += "org.apache.spark" %% "spark-sql" % buildType.sparkVersion % "provided",
       assembly / mainClass := Some("io.treeverse.examples.List"),
-    )
-
-lazy val spark2Type = new BuildType("247", scala211Version, "2.4.7", "0.9.8", "2.7.7")
-lazy val spark3Type = new BuildType("301", scala212Version, "3.0.1", "0.10.11", "2.7.7")
-
-lazy val core2 = generateCoreProject(spark2Type)
-lazy val core3 = generateCoreProject(spark3Type)
-lazy val examples2 = generateExamplesProject(spark2Type).dependsOn(core2)
-lazy val examples3 = generateExamplesProject(spark3Type).dependsOn(core3)
+    ).dependsOn(core)
 
-lazy val root = (project in file(".")).aggregate(core2, core3, examples2, examples3)
+lazy val root = (project in file(".")).aggregate(core, examples).disablePlugins(PublishMore)
```

Comment:
Why do we disable `PublishMore`?  Isn't `root / publish / skip := true` enough, and/or can't we do something similar with config to sbt-publish-more?

Reply:
Unfortunately the plugin it doesn't work well with sub-projects, so I had to disable it for the root project. Otherwise it also tries to publish it.